treeherder/etl/perf_data_adapters.py

```diff
@@ -110,7 +113,7 @@ def _calculate_summary_data(job_id, result_set_id, push_timestamp, results):
 
     @staticmethod
     def _calculate_test_data(job_id, result_set_id, push_timestamp,
-                             replicates):
+                             replicates, summary):
```

Comment:
Likewise, I'd rather keep this method focused on calculating stuff where it's still needed. If necessary create a new method to transform the summary data provided by talos into the format we want and then conditionally call it instead. I think that'll save a lot of if/else logic, as well as being less confusing genrally.